src/prof_recent.c

```diff
@@ -446,8 +455,35 @@ prof_recent_alloc_max_ctl_write(tsd_t *tsd, ssize_t max) {
 	return old_max;
 }
 
+ssize_t
+prof_recent_alloc_max_ctl_write(tsd_t *tsd, ssize_t max) {
+	malloc_mutex_lock(tsd_tsdn(tsd), &prof_dump_mtx);
+	ssize_t old_max = prof_recent_alloc_max_write_locked(tsd, max);
+	malloc_mutex_unlock(tsd_tsdn(tsd), &prof_dump_mtx);
+	return old_max;
+}
+
+static prof_recent_t *
+prof_recent_alloc_swap_and_clear_locked(tsd_t *tsd, ssize_t *dump_count,
```

Comment:
nit: I see why you have `clear` in the name, however it's a bit confusing (we are not really releasing the records, just stashed them in a separate list). Maybe a _prepare and _finish pair?